doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,80 @@
+expected by the kernel.
+
+For instance, the notation mechanisms reflects the eponym mathematical
+practice and allows to write :n:`1 + 2 * 2` instead of the above
+term. Those mechanisms range from simple :ref:`Abbreviations` to full
+fledged :ref:`Notations` with user defined :ref:`syntaxes
+<ReservingNotations>`. Multiple interpretations can be given to the
+same syntax in different contexts thanks to the :ref:`scope
+<Scopes>` mechanism. For instance :n:`(1 + 2 * 2)%N` could denote the
+above natural number expression while :n:`(1 + 2 * 2)%Z` could denote
+a similar expression with integers.
+
+In order to take the best part of all these preprocessing mechanisms,
+one needs a basic understanding of the multiple steps needed to
+transform an inputed term (possibly with notations) into the valid
+Gallina term which Rocq will ultimately use internally. When inputing
+a term to Rocq, it goes through several successive steps:
+
+* During :gdef:`lexing` the input is split into a sequence of tokens,
+  for instance the string ``"1 + 2 * 2"`` will be split into the
+  sequence of five tokens `'1' '+' '2' '*' '2'`. A set of :ref:`basic
+  tokens <lexical-conventions>` are predefined and can be extended, in
+  particular by :ref:`reserving notations <ReservingNotations>`.
+
+* During :gdef:`parsing` the previous stream is given a tree-like
+  structure, for instance here
+
+  .. code-block:: text
+     :name: after-parsing
+
+             +
+            / \
+           1   *
+              / \
+             2   2
+
+  The parsed grammar can be modified by :ref:`reserving notations
+  <ReservingNotations>`.
+
+* During :gdef:`notation interpretation` each syntactic element gets
+  translated to a term, for instance :n:`1` can be interpreted as the
+  natural number :n:`S O` then :n:`2` is interpreted as :n:`S (S O)`,
+  then :n:`2 * 2` as :n:`Nat.mul (S (S O)) (S (S O))` and finally our
+  whole term as :n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`. The
+  same expression can be given multiple interpretations in various
+  contexts thanks to :ref:`Scopes`.
+
+* Finally, :gdef:`elaboration` (also called :gdef:`pretyping` in the
+  source code of Rocq) can use the various mechanisms described in
+  this section to fill gaps (for instance with :ref:`implicit
+  arguments <ImplicitArguments>`, :ref:`canonical structures
+  <canonicalstructures>` or :ref:`typeclasses`) or fix terms (for
+  instance with :ref:`coercions`) to obtain fully detailled terms in
+  the :ref:`Core Language <core-language>`.
+
+For each sentence, coq performs these steps successively and
+independently. Once a step is completed, there is no going back. Then,
+the result goes through the type checking phases discussed in
+:ref:`previous chapter <core-language>`.
+The first three phases (:term:`lexing`, :term:`parsing` and
+:term:`notation interpretation`) are sometimes jointly called the
+syntax interpretation (ot synterp for short) phase, whereas
```

Comment:
notation interpretation / intern is not part of synterp

Reply:
synterp is parsing plus the minimum work to be able to parse the following commands (typically notation reservation + import of reserved notations + enough module handling to resolve what is imported in Import + plugin loading)

Reply:
So it's only lexing and parsing? Sorry, my understanding of synterp/interp remains pretty poor.

Reply:
IINM "synterp" is a term used by developers.  Why do users need to know the term?  We shouldn't define useless terms--it's just more mental clutter.

Reply:
It's not so clear cut, the distinction is also of interest to plugin developers and users of extension languages. Since it's indeed not very useful for casual users, I isolated it in a specific note.